src/ConnectedFieldArray.js

```diff
@@ -23,7 +24,7 @@ const createConnectedFieldArray = ({
 
   class ConnectedFieldArray extends Component {
     shouldComponentUpdate(nextProps) {
-      return size(this.props.value) !== size(nextProps.value)
+      return shallowCompare(this, nextProps)
```

Comment:
@erikras Are you saying something like this? [Link](https://github.com/erikras/redux-form/pull/940/files#diff-c51b679ed2957be9891b8e3164a067b1R312)